File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<int,int> map;
        int ans=0;
        int j =0;
        for(int i = 0; i < s.size(); i++)
        {
            if(map.find(s[i]) != map.end())
                j = max(j, map[s[i]]);
            ans = max(ans, i - j + 1);
            map[s[i]] = i+1;
        }
        return ans;
    }
};

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        start = 0
        maxSol = 0
        seen = dict()
        for charInd in range(len(s)):
            if s[charInd] in seen:
                start = max(seen[s[charInd]],start)
            seen[s[charInd]] = charInd + 1
            maxSol = max(maxSol, charInd - start + 1)
        return maxSol
                